cmd/osv-scanner/fix/main.go

```diff
@@ -196,26 +201,42 @@ func action(ctx *cli.Context) error {
 		}
 		cl, err := client.NewNpmRegistryClient(workDir)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		opts.Client.DependencyClient = cl
 	}
 
 	if opts.Manifest != "" {
 		rw, err := manifest.GetManifestIO(opts.Manifest)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		opts.ManifestRW = rw
 	}
 
 	if opts.Lockfile != "" {
 		rw, err := lockfile.GetLockfileIO(opts.Lockfile)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		opts.LockfileRW = rw
 	}
 
-	return fmt.Errorf("not implemented")
+	if !ctx.Bool("non-interactive") {
+		return nil, fmt.Errorf("not implemented")
+	}
+
+	// TODO: This isn't what the reporter is designed for.
+	// Only using r.Infof() and r.Errorf() to print to stdout & stderr respectively.
+	r := reporter.NewTableReporter(stdout, stderr, reporter.InfoLevel, false, 0)
+	numUpgrades := ctx.Int("apply-top")
+
+	switch ctx.String("strategy") {
+	case "relock":
+		return r, autoRelock(ctx.Context, r, opts, numUpgrades)
+	case "in-place":
+		return r, autoInPlace(ctx.Context, r, opts, numUpgrades)
+	}
+
+	panic("no entrypoint for cli flags")
```

Comment:
Clarify this error being strategy is invalid, and therefore is out of sync with the check under the strategy flag.